wikihow2zim/scraper.py

```diff
@@ -29,9 +38,73 @@ def __init__(self, **kwargs):
         for option in options:
             setattr(self, option, kwargs[option])
 
+        if self.output_dir is None:
+            self.output_dir = pathlib.Path(os.getcwd()).joinpath("output")
+
+        pathlib.Path(self.output_dir).mkdir(parents=True, exist_ok=True)
+
         self.url = URLS[self.language]
 
+        # jinja2 environment setup
+        self.env = Environment(
+            loader=FileSystemLoader(str(self.templates_dir)), autoescape=True
+        )
+
+        # Set up the output zim file:
+        fpath = pathlib.Path(self.output_dir).joinpath("output.zim")
+        self.creator = Creator(filename=fpath).set_mainpath("Home")
```

Comment:
We never create nor start the Zim Creator in `__init__()` because libzim's a C++ extension and that comes with a few constraints ; mostly in error reporting.
For that reason, you'll want to create and start it in your `run()` method, after you're done checking inputs.